controllers/nodeagent.go

```diff
@@ -369,9 +371,35 @@ func (r *DPAReconciler) customizeNodeAgentDaemonset(dpa *oadpv1alpha1.DataProtec
 	if ds.Spec.RevisionHistoryLimit == nil {
 		ds.Spec.RevisionHistoryLimit = pointer.Int32(10)
 	}
+
+	if err := r.applyUnsupportedNodeAgentServerArgsOverride(dpa, nodeAgentContainer); err != nil {
```

Comment:
I would this function along container changes 

would refactor line 305 to be like this
```go

	// fetch nodeAgent container in order to customize it
	for _, container := range ds.Spec.Template.Spec.Containers {
		if container.Name == common.NodeAgent {
			// append certs volume mount
			container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
				Name:      "certs",
				MountPath: "/etc/ssl/certs",
			})
			// append PodConfig envs to nodeAgent container
			if useResticConf {
				if dpa.Spec.Configuration.Restic.PodConfig != nil && dpa.Spec.Configuration.Restic.PodConfig.Env != nil {
					container.Env = common.AppendUniqueEnvVars(container.Env, dpa.Spec.Configuration.Restic.PodConfig.Env)
				}
			} else if dpa.Spec.Configuration.NodeAgent.PodConfig != nil && dpa.Spec.Configuration.NodeAgent.PodConfig.Env != nil {
				container.Env = common.AppendUniqueEnvVars(container.Env, dpa.Spec.Configuration.NodeAgent.PodConfig.Env)
			}

			// append env vars to the nodeAgent container
			container.Env = common.AppendUniqueEnvVars(container.Env, proxy.ReadProxyVarsFromEnv())

			container.SecurityContext = &corev1.SecurityContext{
				Privileged: pointer.Bool(true),
			}

			imagePullPolicy, err := common.GetImagePullPolicy(getVeleroImage(dpa))
			if err != nil {
				r.Log.Error(err, "imagePullPolicy regex failed")
			}

			container.ImagePullPolicy = imagePullPolicy
			setContainerDefaults(&container)

			if err := r.applyUnsupportedNodeAgentServerArgsOverride(dpa, &container); err != nil {
				return nil, err
			}
			break
		}
	}
```

Reply:
actually, I think my suggestion does not work "The analyzer reports instances of writes to struct fields and arrays that are never read. Specifically, when a struct object or an array is copied, its elements are copied implicitly by the compiler, and any element write to this copy does nothing with the original object." Ref https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/unusedwrite

would need to be something like this
```go
	// fetch nodeAgent container in order to customize it
	for index, container := range ds.Spec.Template.Spec.Containers {
		if container.Name == common.NodeAgent {
			nodeAgentContainer := &ds.Spec.Template.Spec.Containers[index]
			// append certs volume mount
			nodeAgentContainer.VolumeMounts = append(nodeAgentContainer.VolumeMounts, corev1.VolumeMount{
				Name:      "certs",
				MountPath: "/etc/ssl/certs",
			})
			// append PodConfig envs to nodeAgent container
			if useResticConf {
				if dpa.Spec.Configuration.Restic.PodConfig != nil && dpa.Spec.Configuration.Restic.PodConfig.Env != nil {
					nodeAgentContainer.Env = common.AppendUniqueEnvVars(nodeAgentContainer.Env, dpa.Spec.Configuration.Restic.PodConfig.Env)
				}
			} else if dpa.Spec.Configuration.NodeAgent.PodConfig != nil && dpa.Spec.Configuration.NodeAgent.PodConfig.Env != nil {
				nodeAgentContainer.Env = common.AppendUniqueEnvVars(nodeAgentContainer.Env, dpa.Spec.Configuration.NodeAgent.PodConfig.Env)
			}

			// append env vars to the nodeAgent container
			nodeAgentContainer.Env = common.AppendUniqueEnvVars(nodeAgentContainer.Env, proxy.ReadProxyVarsFromEnv())

			nodeAgentContainer.SecurityContext = &corev1.SecurityContext{
				Privileged: pointer.Bool(true),
			}

			imagePullPolicy, err := common.GetImagePullPolicy(getVeleroImage(dpa))
			if err != nil {
				r.Log.Error(err, "imagePullPolicy regex failed")
			}

			nodeAgentContainer.ImagePullPolicy = imagePullPolicy
			setContainerDefaults(nodeAgentContainer)

			if err := r.applyUnsupportedNodeAgentServerArgsOverride(dpa, nodeAgentContainer); err != nil {
				return nil, err
			}
			break
		}
	}
```

Reply:
The proposed implementation is nicer, but it does change one condition where the `nodeAgentContainer` may be nil based on the reference from `ds.Spec.Template.Spec.Containers` I am unsure if it's valid case, but does change a bit how the crash may occur in the application.

Reply:
I think that will never happen

if container is nil, calling container.Name would break

Reply:
Done